### PR TITLE
DOC: Fix docs related to `n_jobs`

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR fixes an issue in which the docs for classes that are not part of original scikit-learn were not setting the docs for `n_jobs` correctly and showing warnings about them, due to to three reasons:
* Using internal "glossary" link, which is from original scikit-learn but not present in sklearnex.
* Referencing objects from joblib without linking to them the way scikit-learn does.
* The decorator adding the parameter more than once to the docstrings if the keyword "Attributes" appeared more than once.

Before:
image
(note that glossary and backend are not clickable)

After:
image

---

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

**Performance**

Not applicable.

### CODE REVIEW

File: doc/sources/conf.py [hunk 1]
```diff
@@ -68,8 +68,23 @@
     "sphinx_design",
     "sphinx_copybutton",
     "sphinx.ext.napoleon",
+    "sphinx.ext.intersphinx",
 ]
 
+intersphinx_mapping = {
```
Comment (icfaust) [inline doc/sources/conf.py:74]: Wishful thinking: is there any way to get this from sklearn itself? I imagine this was a maintenance burden we just didn't track until your good work.

**Review by icfaust — COMMENTED** [review]
otherwise looks good to me, just wondered something small.

File: doc/sources/conf.py (hunk 1 above)
Comment (david-cortes-intel) [reply doc/sources/conf.py:74]: I don't think so, because this is only part of the docs `conf.py` file, which is not part of the python package.

**Review by icfaust — APPROVED** [review]
